### PR TITLE
[FIX]: Fixing support for overriding the SCSS variables

### PR DESCRIPTION
**Pull request by jrubins**
## Context

I was updating our project's `simple-line-icons` dependency to the `v2.5.2` version and when I went to test our build, I noticed that it was failing because our overriding of the `$simple-line-font-path` variable wasn't being applied.

When I looked at the changes made recently, I saw that the `!default` declarations had been dropped from the built SCSS file. In other words, older versions (specifically version `v2.4.1`) had the following:
```
$simple-line-font-path: "../fonts/" !default;
```
But in `v2.5.2` that line looks like:
```
$simple-line-font-path: "../fonts/";
```
([See the full comparison to see these changes.](https://github.com/thesabbir/simple-line-icons/compare/2.4.1...2.5.2#diff-65f17b06fdca5b53e248dace75a2a33cL1))

## Changes

Looking at how the SCSS file was being generated, I saw that the `less2sass.js` script was used for this transformation. There doesn't seem to be support there for default variables because [LESS doesn't have a specific default declaration and instead just relies on ordering](http://lesscss.org/features/#variables-feature-default-variables). Therefore, I didn't see anything in the `less2sass` code that would support adding this default declaration easily (because there's no good way to determine what should be a default variable vs what shouldn't be).

So the changes I made in this PR are to modify the `less2sass.js` script to support adding the `!default` declaration for the variables that we want to be overridable.

## Testing

I used my forked repository in our project and it seemed to resolve the issue (i.e. the variables were overridable again).

### CODE REVIEW

File: docs/index.html [hunk 1]
```diff
@@ -23,7 +23,7 @@
                 <span class="icon-bar"></span>
             </button>
             <a class="navbar-brand" href="https://github.com/thesabbir/simple-line-icons">Simple Line Icons </a>
-            <span class="version">2.5.0</span>
+            <span class="version">2.5.2</span>
```
Comment (jrubins) [inline docs/index.html:26]: I didn't intentionally make this change, but when I ran the `build` script this got updated. It seems to be an appropriate change though?

File: scripts/less2sass.js [hunk 2]
```diff
@@ -91,9 +92,35 @@ Less2Sass.prototype.convertInterpolatedVariables = function() {
...
+    // for "simple-line-icons"-specific logic for creating default variables
+    // (because LESS doesn't have the concept of default variable declarations).
+    //
+    // For example, if we have a LESS variable that looked like:
+    //
+    //      @simple-line-font-path: "../fonts/";
+    //
+    // The "varDeclaration" group would be 'simple-line-font-path: "../fonts/"' and the varName
+    // group would be "simple-line-font-path".
+    var atRegex = /@(?!(media|import|mixin|font-face|keyframes)(\s|\())(?<varDeclaration>(?<varName>([^:]*))[^;]*)/g;
```
Comment (jrubins) [inline scripts/less2sass.js:107]: I'm no RegEx expert, but I used regex101.com to validate that I was selecting what I desired:



As mentioned in the comment, the `varDeclaration` and `varName` capturing groups are the additions made to run the below logic of defaulting only the variables that we want to be overridable.

**Comment by garbinmarcelo** [comment]
I opened an issue with this problem and ended up generating a pull request without seeing yours. I believe that the author can close mine in this case. Thank you

File: scripts/less2sass.js (hunk 2 above)
Comment (mruz) [reply scripts/less2sass.js:107]: What if we add `// default` comment and match default variables by this comment? https://regex101.com/r/zsu4Ba/1
Screenshot_20200805_101524

Comment (jrubins) [reply scripts/less2sass.js:107]: That sounds like a good idea to me! I'll update this weekend.

**Comment by thesabbir** [comment]
Please resolve conflict and do a build.
I'll merge this.

**Comment by thesabbir** [comment]
@jrubins  thanks for this pr 👍

File: scripts/less2sass.js [hunk 3]
```diff
@@ -91,9 +92,33 @@ Less2Sass.prototype.convertInterpolatedVariables = function() {
...
+    // creating default variables (because LESS doesn't have the concept of default variable declarations).
+    //
+    // For example, if we have a LESS variable that looked like:
+    //
+    //      @simple-line-font-path: "../fonts/"; // default
+    //
+    // The "varDeclaration" group would be 'simple-line-font-path: "../fonts/"', the varName
+    // group would be "simple-line-font-path" and the isDefault group would have content (indicating
+    // that this should be output as a default variable).
+    var varRegex = /@(?!(media|import|mixin|font-face|keyframes)(\s|\())(?<varDeclaration>(?<varName>([^:]*))[^;]*);(?<isDefault>\s*\/\/\s*default)?/g;
```
Comment (jrubins) [inline scripts/less2sass.js:108]: @mruz I had to make this a capturing group which I called `isDefault` and I had to make it optional. This is to support transforming variables that aren't supposed to be default variables. For example, if we had the following:
```
@defaultVar: "value"; // default
@nonDefaultVar: "value";
```
we would want the output of:
```
$defaultVar: "value" !default;
$nonDefaultVar: "value";
```

Comment (thesabbir) [reply scripts/less2sass.js:108]: creative
 :D

**Pull request merged**